byzcoin/struct.go

```diff
@@ -1,12 +1,531 @@
 package byzcoin
 
 import (
+	"bytes"
+	"encoding/binary"
 	"errors"
+	"math/big"
+	"sort"
 	"sync"
+	"time"
 
+	bolt "github.com/coreos/bbolt"
 	"github.com/dedis/cothority/skipchain"
+	"github.com/dedis/onet"
+	"github.com/dedis/protobuf"
 )
 
+const defaultMaxSize = 2 * 1024 * 1024 * 1024 // 2GB
+const versionLength = 64 / 8                  // bytes
+const prefixLength = 32                       // bytes
+const cleanThreshold = 0.8
```

Comment:
why do we need both a max size and a threshold? isn't maxsize by itself enough?

Reply:
Max size is the limit and the threshold tells how much will be cleaned if the storage goes above max size. My goal was to prevent multiple small rounds of cleaning.